cmd/otelcol/main_test.go

```diff
@@ -142,3 +148,34 @@ func TestSetMemoryLimit(t *testing.T) {
 
 	HelperTestSetMemoryLimit("200", t)
 }
+
+func TestConfigPathAndYamlEnvVarsTogetherNotAllowed(t *testing.T) {
+	expectedErr := fmt.Sprintf("Specifying env vars %s and %s simultaneously is not allowed", configEnvVarName, configYamlEnvVarName)
+	stdErr, exitCode := getExecErr(t, "TestConfigPathAndYamlEnvVarsTogetherNotAllowed", func(t *testing.T) {
+		// Setting both config file path and config YAML env vars
+		os.Setenv(configYamlEnvVarName, "service:  \\nextensions:  \\n- health_check")
+		os.Setenv(configEnvVarName, "path/to/config/file")
+		setConfigSource()
+	})
+
+	require.NotZero(t, exitCode)
+	assert.Containsf(t, stdErr, expectedErr, "Standard error does not contain the expected error")
+}
+
+// getExecErr should be called once in a test at the beginning and the failing code placed in func failingCode.
+// getExecErr runs the test in a subprocess then returns the output of executing failingCode.
+func getExecErr(t *testing.T, testName string, failingCode func(*testing.T)) (string, int) {
```

Comment:
This doesn't seem like a standard go testing convention.  Why would running a test as a subprocess be necessary, especially one calling itself?

Reply:
Not seeing how this is resolved.  I have only seen equivalent functionality in multiprocessing plugins for test frameworks.  Please write a standard test case without this functionality, and use the integration testing testutils if you want to run a collector process: https://github.com/signalfx/splunk-otel-collector/tree/main/tests